### ISSUE TITLE
UISwitch not responding to taps or swipes

### ISSUE THREAD

**Issue by billburgess** [kif-framework/KIF#1039]
This issue is with Xcode 9.2 only. I'm not having any issues with anything prior to this version.

I have a ton of tests that are failing in Xcode 9.2 on calls to -tapWithAccessibilityLabel: as well as -swipeViewWithAccessibilityLabel:. I have some calls to this that do seem to working, but a lot are not. I'm setting things the same everywhere, but I can't get it to find the element and interact with it. It is also not finding -waitForViewWithAccessibilityLabel either.

Specifically, the call to tap or swipe succeeds as far as pass/fail goes, but the action isn't being performed. What ends up happening is that the next line in the test will fail as it is waiting on the action to do something (present alert, change screen, etc) and that isn't happening. I can manually tap or swipe while it is waiting and the remainder of the test will pass. Just no actions being performed on UISwitch.

This issue seems to be limited only to UISwitch. Has anyone else had this issue? Any thoughts on things to try? I've been running my full suite on the previous version of Xcode, but soon that isn't going to work and I don't have a fix for any of these failing tests.

**Comment by billburgess** [comment]
I noticed the following output log while it is waiting for the step after the switch toggle.

```
2018-01-12 14:17:28.104215-0600 simpleinout[70518:4124089] [Accessibility] Executing an update request
2018-01-12 14:17:28.104414-0600 simpleinout[70518:4124089] [Accessibility] Ignoring an update request
2018-01-12 14:17:28.109804-0600 simpleinout[70518:4124089] [Accessibility] Update request succeeded
2018-01-12 14:17:28.865561-0600 simpleinout[70518:4124089] [Accessibility] Executing an update request
2018-01-12 14:17:28.866336-0600 simpleinout[70518:4124089] [Accessibility] Update request succeeded
2018-01-12 14:17:29.611836-0600 simpleinout[70518:4124089] [Accessibility] Executing an update request
2018-01-12 14:17:29.622261-0600 simpleinout[70518:4124089] [Accessibility] Update request succeeded
2018-01-12 14:17:30.268544-0600 simpleinout[70518:4124089] [Accessibility] Executing an update request
2018-01-12 14:17:30.270246-0600 simpleinout[70518:4124089] [Accessibility] Update request succeeded
```

**Comment by mikelupo** [comment]
Stab in the dark. 
I have a PR up that’s a few months old. Can you try with that?

**Comment by billburgess** [comment]
I'll give it a try and report back.

**Comment by billburgess** [comment]
I'm not able to get the switch to work with -tapView... but does seem to work with your PR for -swipeView... I'm running my full suite now against your changes in branch 'fixForIssue1007'. I'll report back my final run.

**Comment by billburgess** [comment]
Using your pull request, I was able to get everything to pass with one caveat. UISwitch events will only respond to -swipeViewWithAccessibilityLabel. Any place I had a -tapViewWithAccessibilityLabel the command would pass, but the event would not trigger. Refactoring to use -swipe seems to have fixed the issue with your changes.

I still think there is work to do to get things working with -tap since that is still a valid action.

**Comment by billburgess** [comment]
Not sure if this is related to your PR branch... but my suite is slow as hell. Like 2x slower than usual. I have the window layer dialed up to 100 for ludicrous mode and things usually fly pretty good. But not so much using this latest branch.

**Comment by justinseanmartin** [comment]
@billburgess - Is this only after your tests are running for a while? I wonder if you're encountering this issue:
https://forums.developer.apple.com/thread/94323
https://forums.developer.apple.com/thread/89616

Our fix was tweeted here:
https://twitter.com/kyleve/status/941151290989481984

**Comment by billburgess** [comment]
@justinseanmartin No, I don't think so. I have the problem even when running 1 test at a time. I had a similar issue before with the suite just choking the memory out, but solved by breaking the suite into multiple targets so they run faster and free things up when done.

**Comment by justinseanmartin** [comment]
This only happens on Xcode 9.2 though and not 9.1? Is this on device or in simulator? Have you looked in instruments to see what is chewing up CPU or memory?

**Comment by billburgess** [comment]
Only on Xcode 9.2. No issues on 9.1. Simulator, haven't tested on live devices. CPU and memory isn't really an issue, just indicates that the tap or swipe happened and moves on to next step, but the switch isn't moving which usually fails the test since the next thing isn't set or moved.

**Comment by justinseanmartin** [comment]
Is it just a `UISwitch` or a special subclass? Do you see the same issues when running `SpecificControlTests testTogglingASwitch`? You could try putting a loop around it and running it 100 times to see if that exacerbates it. 

I'm just trying to narrow down to a scenario that I could investigate locally.

**Comment by billburgess** [comment]
Just a boring UISwitch, no subclassing at all. I tried recreating the issue in a new project but doesn't seem to be a problem on just single taps/swipes. There's some other combination of events going on that I can't quite figure out. I'll try and put some of these combinations and try to brute force it and see what happens.

I did use the PR branch from @mikelupo and my tests ran like they should, albeit a little slower than I'd like.

**Comment by billburgess** [comment]
I tried a brute force test on this by tapping on, swiping off 100 times. Still seems to work by itself. I can't figure out why these are a problem in my suite.

Also having issues with them using @mikelupo changes as well.

**Comment by billburgess** [comment]
I think I got things working well enough. I'm still not entirely sure what my suite/tests are doing that are affected by 9.2 so much, but I think I'm in a state that I can close this issue and deal with it on my own. I can't recreate the issue standalone so it is probably safe to rule KIF out.

If anyone stumbles upon this, here is what I did to get things working better.

- Better cleanup between runs. Killing off views that might not be releasing when changing out my window rootViewController. So less views hanging around between tests that may confuse any lookups for elements.
- The couple of places still plagued by the UISwitch problem I converted from using the -tapView and -swipeView to the specific setting of the switch by using the -setOn method. Manually setting the switch on works every time and isn't affected by the tap/swipe issues.

Thanks for looking into the issue, I'm going to call this one closed due to testing cleanup on my end.
